databuilder/publisher/neptune_csv_publisher.py

```diff
@@ -0,0 +1,177 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+import datetime
+import logging
+import os
+import time
+from io import BytesIO
+from os import listdir
+from os.path import isfile, join
+from typing import List, Tuple
+
+from amundsen_gremlin.neptune_bulk_loader.api import NeptuneBulkLoaderApi, NeptuneBulkLoaderLoadStatusErrorLogEntry
+from boto3.session import Session
+from pyhocon import ConfigTree
+
+from databuilder.publisher.base_publisher import Publisher
+
+LOGGER = logging.getLogger(__name__)
+
+
+class NeptuneCSVPublisher(Publisher):
+    """
+    This Publisher takes two folders for input and publishes to Neptune.
+    One folder will contain CSV file(s) for Node where the other folder will contain CSV file(s) for Relationship.
+
+    This publisher uses the bulk api found in
+    https://github.com/amundsen-io/amundsengremlin/blob/master/amundsen_gremlin/neptune_bulk_loader/api.py
+
+    which is a client for the the api found
+    https://docs.aws.amazon.com/neptune/latest/userguide/bulk-load.html
+    https://docs.aws.amazon.com/neptune/latest/userguide/bulk-load-tutorial-format-gremlin.html
+    """
+
+    # A directory that contains CSV files for nodes
+    NODE_FILES_DIR = 'node_files_directory'
+    # A directory that contains CSV files for relationships
+    RELATION_FILES_DIR = 'relation_files_directory'
+
+    # --- AWS CONFIGURATION ---
+    # S3 bucket to upload files to
+    AWS_S3_BUCKET_NAME = 'bucket_name'
+    # S3 location where amundsen data can be exported to and Neptune can access
+    AWS_BASE_S3_DATA_PATH = 'base_amundsen_data_path'
+
+    NEPTUNE_HOST = 'neptune_host'
+
+    # AWS CONFIGURATION
+    AWS_REGION = 'aws_region'
+    AWS_ACCESS_KEY = 'aws_access_key'
+    AWS_SECRET_ACCESS_KEY = 'aws_secret_acces_key'
+    AWS_SESSION_TOKEN = 'aws_session_token'
+    AWS_IAM_ROLE_NAME = 'aws_iam_role_name'
+
+    FAIL_ON_ERROR = "fail_on_error"
+    STATUS_POLLING_PERIOD = "status_polling_period"
+
+    def __init__(self) -> None:
+        super(NeptuneCSVPublisher, self).__init__()
+
+    def init(self, conf: ConfigTree) -> None:
+        self._boto_session = Session(
```

Comment:
I do not think they should. The api used by the NeptuneCSVPublisher are used in a different context than the api's used by the NeptuneSessionClient. I feel like it could be combined if you feel strongly. But I feel like it might break the abstraction of the client if the two should be blended together.

Reply:
No strong feelings, totally fine to leave separate if we feel they may diverge more in the future -- just noticed that a lot of the code reading from config and forming connection string was similar and wondered.